src/main/java/sparksoniq/jsoniq/compiler/translator/expr/flowr/ForClause.java

```diff
@@ -56,6 +57,15 @@ public void setPreviousClause(FlworClause previousClause) {
         forVariables.get(0).previousClause = this.previousClause;
     }
 
+    @Override
+    public void initHighestExecutionMode() {
+        // call isDataFrame on the last forVariable
+        this._highestExecutionMode =
+            forVariables.get(forVariables.size() - 1).getHighestExecutionMode().isDataFrame()
+                ? ExecutionMode.DATAFRAME
+                : ExecutionMode.LOCAL;
```

Comment:
Is there a reason why not just

this._highestExecutionMode = forVariables.get(forVariables.size() - 1).getHighestExecutionMode()

?

Reply:
No specific reason, just an oversight :) Good catch!